### ISSUE TITLE
Bug: [no-non-null-assertion] Fix puts ? on wrong line when member access is on next line

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8179]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gNyhAriAFzyYwCWKA5gNwBQduBIAdALZQZgAWAFAPRQARmH4BKAIR148FgDNoGXmJpA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1s6e4XloBDZMkTR8lDk3RQx0DtEjgwAXxAqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=false

### Repro Code

```TypeScript
declare const value: string;

value.match(/abc/)!
  .flat();
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-non-null-assertion": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true
  }
}
```

### Expected Result

Auto-fixing should result in something like:

```ts
declare const value: string;

value.match(/abc/)
  ?.flat();
```

### Actual Result

```ts
declare const value: string;

value.match(/abc/)?
  .flat();
```

### Additional Info

Interestingly, it does work well when the member access is computed:

```ts
declare const value: string;

value.match(/abc/)!
  ['flat']();
```
